agent/indy_catalyst_agent/messaging/credentials/manager.py

```diff
@@ -79,61 +90,99 @@ def context(self) -> RequestContext:
             + f"{connection_id}"
         )
 
+        source_credential_exchange = None
+
         if source_credential_exchange_id:
 
+            # The cached credential exchange ID may not have an associated credential
+            # request yet. Wait up to 30 seconds for that to be populated, then
+            # move on and replace it as the cached credential exchange
+
+            lookup_start = time.perf_counter()
+            while True:
```

Comment:
What is the scenario where this is necessary?

Reply:
@nrempel I was finding when (f.ex) 10 'send' requests are received at once, it would kick off 10 credential requests for the same credential definition. This change prevents that from happening and speeds it up a lot.

Reply:
Ok gotcha - what about when there is only one credential per credential request? Would that wait 30 seconds before completing the issuance?

Wondering if this is beneficial in our use case - but not as ideal for the general case. Also for our case, there might be some slowdown for the first few credentials, but generally, once the cache is hot, we should still see improvements for significant numbers (more than 10)

If that's the case, do you think this optimization is beneficial for the performance test script but not optimal for the general case?

Let me know if I'm missing something, but I think we could leave this out.

Reply:
@nrempel There shouldn't be any impact when there's only one credential, as it won't find a cached credential_exchange_id to wait for. I don't think there's a downside to the change.

Reply:
👍🏼